app/models/debt_balance.rb

```diff
@@ -100,6 +100,10 @@ def self.search(search)
   private
 
   def start_pay_date
+    if self.debt_id == nil
+    errors.add(:debt, "debt must not be null")
```

Comment:
remove the debt in your message. otherwise the error on the application will be "Debt debt must not be null". Also think about the user :) they will not know/care about null ;)